File: src/main/java/bot/util/CommandsUtil.java
package bot.util;

import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
import org.telegram.telegrambots.meta.api.objects.Message;
import org.telegram.telegrambots.meta.api.objects.Update;

import java.time.ZonedDateTime;

public class CommandsUtil {
    public static String getChatId(Update update){
        String chatId;
        Message message = update.getMessage();
        if (message != null) chatId = message.getChatId().toString();
        else chatId = update.getCallbackQuery().getMessage().getChatId().toString();
        return chatId;
    }

    public static void notifyAdmin(Exception e, Update update){
        String chatId = System.getenv("ADMIN");

        SendMessage sendMessage = new SendMessage();
        sendMessage.setChatId(chatId);

        if (update.getMessage().getText() != null) sendMessage.setText(String.format("User {%s} has exception {%s} on request {%s} at {%s}", getChatId(update), e.toString(), update.getMessage().getText(), ZonedDateTime.now()));
        else sendMessage.setText(String.format("User {%s} has exception {%s} on request {%s} at {%s}", getChatId(update), e.toString(), update.getMessage().toString(), ZonedDateTime.now()));
    }
}

File: src/main/java/bot/util/CommandsUtil.kt
package bot.util

import org.telegram.telegrambots.meta.api.methods.send.SendMessage
import org.telegram.telegrambots.meta.api.objects.Update
import java.time.ZonedDateTime

object CommandsUtil {
    fun getChatId(update: Update): String {
        return update.message?.chatId?.toString() ?: update.callbackQuery.message.chatId.toString()
    }

    fun notifyAdmin(e: Exception, update: Update) {
        val chatId = System.getenv("ADMIN")
        val sendMessage = SendMessage()
        sendMessage.chatId = chatId
        if (update.message.text != null) sendMessage.text = String.format(
            "User {%s} has exception {%s} on request {%s} at {%s}",
            getChatId(update),
            e.toString(),
            update.message.text,
            ZonedDateTime.now()
        ) else sendMessage.text =
            String.format(
                "User {%s} has exception {%s} on request {%s} at {%s}",
                getChatId(update),
                e.toString(),
                update.message.toString(),
                ZonedDateTime.now()
            )
    }
}